modules/setting/setting.go

```diff
@@ -320,17 +322,20 @@ func NewContext() {
 	AppSubUrlDepth = strings.Count(AppSubUrl, "/")
 
 	Protocol = HTTP
+	protocol = "http"
 	if sec.Key("PROTOCOL").String() == "https" {
 		Protocol = HTTPS
+		protocol = "https"
 		CertFile = sec.Key("CERT_FILE").String()
 		KeyFile = sec.Key("KEY_FILE").String()
 	} else if sec.Key("PROTOCOL").String() == "fcgi" {
 		Protocol = FCGI
+		protocol = "fcgi"
 	}
 	Domain = sec.Key("DOMAIN").MustString("localhost")
 	HttpAddr = sec.Key("HTTP_ADDR").MustString("0.0.0.0")
 	HttpPort = sec.Key("HTTP_PORT").MustString("3000")
-	LocalURL = sec.Key("LOCAL_ROOT_URL").MustString(Protocol + HttpAddr + HttpPort + "/")
+	LocalURL = sec.Key("LOCAL_ROOT_URL").MustString( protocol + HttpAddr + HttpPort + "/")
```

Comment:
Thanks  for  you  advice ,    this  fcgi  also  I  do not  known  how  it  works!  then   I  will  change the HttpPort  to  Domain.